File: 006LinerList/Java/src/Main.java
/**
 * Created by naoki on 2015/07/02.
 */
public class Main {
    public static void main(String [] args){

        MyListAlpha<String> list = new MyListAlpha<String>();

        list.add("h");
        list.add("e");
        list.add("l");
        list.add("l");
        list.add("o");

        System.out.println(list);
        System.out.println(list.size());
        System.out.println(list.get(3));

        MyList<String> l2 = new Cons<>("hello",new Cons<>("World",new Nil<>()));
        System.out.println(l2.get(1).get());
        MyList<String> l3 = l2.add("Test");
        System.out.println(l3.get(3));
        System.out.println(l3);
        MyList<String> l4 = l3.reverse();
        System.out.println(l3.apend(l4));
        MyList<Integer> l5 = l3.map(str -> Integer.valueOf(str.hashCode()));
        System.out.println(l5);

    }
}

File: 006LinerList/Scala/src/main/scala/Main.scala
/**
 * Created by naoki on 2015/07/06.
 */
object Main extends App{

  val l1 = new MyListAlpha[String]()
  val l2 = l1.add("apple").add("an").add("have").add("I")
  println(l2)
  println(l2.get(3))


  val l3 = MyList.apply("I","have","an","apple")
  val l4 = l3.add("test")
  println(l3.get(2))
  println(l4)
  println(l4.reverse)
  println(l4.append(l3))
  println(l4.map(str=>str.hashCode()))
  println(l4.drop(3))
  println(l4.take(2))
  println(l4.drop(3).append(l4.take(3)))

}
